src/R2.jl

```diff
@@ -56,13 +56,15 @@ mutable struct R2Solver{T, V}
   x::V
   gx::V
   cx::V
+  d::V
   output::GenericExecutionStats{T, V}
 end
 
 function R2Solver(nlp::AbstractNLPModel{T, V}) where {T, V}
   x = similar(nlp.meta.x0)
   gx = similar(nlp.meta.x0)
   cx = similar(nlp.meta.x0)
+  d = zero(similar(nlp.meta.x0))
```

Comment:
```suggestion
  d = fill!(0, similar(nlp.meta.x0))
```

Reply:
This gives me an error and I also tried 
fill!(T(0), similar(nlp.meta.x0))

Reply:
Test callback: Error During Test at C:\Users\Farhad\Documents\GitHub\JSOSolvers.jl\test\callback.jl:2
  Got exception outside of a @test
  MethodError: no method matching fill!(::Float64, ::Vector{Float64})
  Closest candidates are:
    fill!(::Array{T, N} where N, ::Any) where T at array.jl:332
    fill!(::BitArray, ::Any) at bitarray.jl:377
    fill!(::SubArray{Bool, var"#s832", var"#s831", Tuple{AbstractUnitRange{Int64}}, L} where {var"#s832", var"#s831"<:BitArray, L}, ::Any) at multidimensional.jl:1421
    ...
  Stacktrace:
    [1] R2Solver(nlp::ADNLPModel{Float64, Vector{Float64}, Vector{Int64}})
      @ JSOSolvers C:\Users\Farhad\Documents\GitHub\JSOSolvers.jl\src\R2.jl:67
    [2] R2(nlp::ADNLPModel{Float64, Vector{Float64}, Vector{Int64}}; kwargs::Base.Iterators.Pairs{Symbol, var"#cb#3"{Vector{Float64}, Vector{Float64}}, Tuple{Symbol}, NamedTuple{(:callback,), Tuple{var"#cb#3"{Vector{Float64}, Vector{Float64}}}}})
      @ JSOSolvers C:\Users\Farhad\Documents\GitHub\JSOSolvers.jl\src\R2.jl:73
    [3] (::var"#1#4"{var"#cb#3"{Vector{Float64}, Vector{Float64}}, ADNLPModel{Float64, Vector{Float64}, Vector{Int64}}})()
      @ Main C:\Users\Farhad\Documents\GitHub\JSOSolvers.jl\test\callback.jl:16
    [4] with_logstate(f::Function, logstate::Any)
      @ Base.CoreLogging .\logging.jl:491
    [5] with_logger(f::Function, logger::Base.CoreLogging.NullLogger)
      @ Base.CoreLogging .\logging.jl:603
    [6] macro expansion
      @ C:\Users\Farhad\Documents\GitHub\JSOSolvers.jl\test\callback.jl:15 [inlined]
    [7] macro expansion
      @ C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Test\src\Test.jl:1151 [inlined]
    [8] top-level scope
      @ C:\Users\Farhad\Documents\GitHub\JSOSolvers.jl\test\callback.jl:3
    [9] include(fname::String)
      @ Base.MainInclude .\client.jl:444
   [10] top-level scope
      @ C:\Users\Farhad\Documents\GitHub\JSOSolvers.jl\test\runtests.jl:10
   [11] include(fname::String)
      @ Base.MainInclude .\client.jl:444
   [12] top-level scope
      @ none:6
   [13] eval
      @ .\boot.jl:360 [inlined]
   [14] exec_options(opts::Base.JLOptions)
      @ Base .\client.jl:261
   [15] _start()
      @ Base .\client.jl:485

Reply:
sorry, it should be `fill!(similar(...), 0)`.